pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -406,6 +406,24 @@ type MonitoringSpec struct {
 	// Interval determines prometheus scrape interval
 	// +optional
 	Interval *metav1.Duration `json:"interval,omitempty"`
+
+	// Ceph exporter configuration
+	// +optional
+	Exporter *CephExporterSpec `json:"exporter,omitempty"`
+}
+
+type CephExporterSpec struct {
+	// Only performance counters greater than or equal to this option are fetched
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=10
+	// +kubebuilder:default=5
+	PerfCountersPrioLimit int `json:"perfCountersPrioLimit,omitempty"`
+
+	// Time to wait before sending requests again to exporter server (seconds)
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=600
+	// +kubebuilder:default=5
```

Comment:
I remember we have issues setting default I API,
cc @BlaineEXE

Reply:
We are using the `kubebuildiner:default` in several places without issue. I think it was an issue with older K8s versions that is no longer a concern.

Reply:
Okay, if it works we are fine, 
Blaine had some suggestions adding default in API is not that good. https://github.com/rook/rook/pull/14435#discussion_r1710034114

Reply:
I haven't tested it in a while, but I have seen that when Rook detects a new CR added, it gets ADDed with the zero value, then quickly gets UPDATEd with the default. Sometimes this results in a double reconcile. I'm not sure if that's still the case or not. 

Even if the scenario I described still exists, the worst that happens is that we apply the zero value, then reconcile again with the default value. As long as the zero value is safe to use, it shouldn't be an issue.

I think it would be good to understand what the zero value means here though. Does zero mean the wait time is (A) actually zero seconds? Is it (B) infinity seconds (i.e. resend disabled)? Does it mean the exporter (C) uses its own internal default? (D) other?

If it's (C), then maybe Rook doesn't need to specify kubebuilder:default at all. If it's (A), there probably isn't an issue. If it's (B), we should make sure with the Ceph team that changing from 0(infinity) to 5 doesn't have problematic edge case behaviors. (B) could risk some exported thing(s) not getting resent at all if the exporter doesn't handle that edge case well.

Reply:
From what I can tell looking at Ceph's code, I think the zero value means zero seconds. 

https://github.com/ceph/ceph/blob/ccdd877abaf707fabc7e0b8d209a421a2f841be0/src/exporter/DaemonMetricCollector.cc#L32-L47

Additionally, if we set 0 specifically, I think it will take 0 to be the value, rather than taking Ceph's internal default of 5s. I'm not absolutely certain, but this is a safe assumption.

I don't see anything in Ceph's code that limits this to 600s. Is there a need/reason to set this maximum? https://github.com/ceph/ceph/blob/51a89906dff333becb40508f6b80029646801f8c/src/common/options/ceph-exporter.yaml.in#L37-L45

Since 0 is a valid setting for Ceph, I think it makes sense to pass the 0 value on from Rook to Ceph if Rook has zero in the CR. We could make this a pointer to an int `*int` instead, so that we can simply not apply the argument if the user leaves this unset, but that might be more work than is truly needed.

I think this (removing the max limit) will be fine. @travisn wdyt? 
```go
	// +kubebuilder:validation:Minimum=0
	// +kubebuilder:default=5
```

Reply:
I just ran a quick test on this branch, and didn't see any duplicate reconcile. Perhaps that was an issue of a previous controller runtime that has since been fixed.